CMakeLists.txt

```diff
@@ -207,22 +207,43 @@ endif()
 
 # If available and wanted, use compiler cache.
 if(LIBCELLML_COMPILER_CACHE)
-  if(WIN32)
-    if(CLCACHE_EXE)
-      set(CLCACHEWRAPPER ${CMAKE_CURRENT_BINARY_DIR}/clcachewrapper)
-      execute_process(COMMAND ${CMAKE_C_COMPILER} /O2 /Fe${CLCACHEWRAPPER} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/clcachewrapper.c
-                      RESULT_VARIABLE RESULT OUTPUT_QUIET ERROR_QUIET)
-      if(RESULT EQUAL 0)
-        set(CMAKE_CXX_COMPILER_LAUNCHER ${CLCACHEWRAPPER})
-      endif()
-    endif()
+  if(BUILDCACHE_EXE)
+    set(CMAKE_CXX_COMPILER_LAUNCHER ${BUILDCACHE_EXE})
   else()
-    if(CCACHE_EXE)
-      set(CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_EXE})
+    if(WIN32)
+      if(CLCACHE_EXE)
+        set(CLCACHEWRAPPER ${CMAKE_CURRENT_BINARY_DIR}/clcachewrapper)
+        execute_process(COMMAND ${CMAKE_C_COMPILER} /O2 /Fe${CLCACHEWRAPPER} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/clcachewrapper.c
+                        RESULT_VARIABLE RESULT OUTPUT_QUIET ERROR_QUIET)
+        if(RESULT EQUAL 0)
+          set(CMAKE_CXX_COMPILER_LAUNCHER ${CLCACHEWRAPPER})
+        endif()
+      endif()
+    else()
+      if(CCACHE_EXE)
+        set(CMAKE_CXX_COMPILER_LAUNCHER ${CCACHE_EXE})
+      endif()
     endif()
   endif()
 endif()
 
+# On Windows, if we are using MSVC then replace some compiler flags.
+if(WIN32 AND MSVC)
```

Comment:
I'm not sure where else you would use `MSVC`.

Reply:
Agreed. See commit 6a452f55.